frontend/src/metabase/browse/databases/BrowseDatabases.tsx

```diff
@@ -52,27 +66,75 @@ export const BrowseDatabases = () => {
-          <DatabaseGrid data-testid="database-browser">
-            {databases.map(database => (
-              <div key={database.id}>
-                <DatabaseCardLink to={Urls.browseDatabase(database)}>
-                  <DatabaseCard>
-                    <Icon
-                      name="database"
-                      color={color("accent2")}
-                      className={CS.mb3}
-                      size={32}
-                    />
-                    <Title order={2} size="1rem" lh="1rem" color="inherit">
+          <BrowseGrid data-testid="database-browser">
+            {databases &&
+              databases.length > 0 &&
+              databases.map(database => (
+                <Link to={Urls.browseDatabase(database)} key={database.id}>
+                  <DatabaseCard
+                    bg="bg-white"
+                    className={cx(CS.rounded, CS.bordered, DB.dbCard)}
+                  >
+                    <Icon name="database" color={color("accent2")} size={32} />
+                    <Title order={2} size="md" lh="md" color="inherit">
                       {database.name}
                     </Title>
                   </DatabaseCard>
-                </DatabaseCardLink>
-              </div>
-            ))}
-          </DatabaseGrid>
+                </Link>
+              ))}
+
+            {isAdmin && (
+              <Link to={newDatabase()} onClick={() => trackAddDatabaseDBList()}>
+                <DatabaseCard
+                  className={cx(CS.rounded, CS.bordered, DB.addCard)}
+                >
+                  <Group spacing="xs">
+                    <CardImageWrapper database={"postgres"} />
+                    <CardImageWrapper database={"mysql"} />
+                    <CardImageWrapper database={"snowflake"} />
+                  </Group>
+                  <div>
+                    <Title order={2} size="md" lh={1} color="inherit">
+                      {t`Add a database`}
+                    </Title>
+                    <Text
+                      color="inherit"
+                      fz="sm"
+                      lh={1}
+                      span
+                    >{t`20+ data connectors. Start exploring in minutes.`}</Text>
+                  </div>
+                </DatabaseCard>
+              </Link>
+            )}
+          </BrowseGrid>
         </BrowseSection>
       </BrowseMain>
     </BrowseContainer>
   );
 };
+
+const CardImageWrapper = ({ database }: { database: string }) => {
+  return (
+    <Box
+      bg="white"
+      h="xl"
+      w="xl"
+      className={CS.rounded}
+      style={{
+        boxShadow:
+          "0px 0px 0px 1px rgba(0, 0, 0, 0.05), 0px 1px 4px 0px rgba(0, 0, 0, 0.10)",
+      }}
+    >
+      <img src={getEngineLogo(database)} alt={t`${database} database logo`} />
+    </Box>
+  );
+};
+
+const DatabaseCard = ({ children, ...props }: StackProps) => {
+  return (
+    <Stack h="8.5rem" justify="space-between" p="lg" {...props}>
```

Comment:
Before this PR, none of the cards had the upper limit so the grid could look weird and misaligned it the database name was long.
![image](https://github.com/user-attachments/assets/2a3ab5b6-06a5-4a57-99fb-fe7a5f9b7f09)


I talked to @rafpaf and we agreed it should be ok to set some reasonable height on these cards.
Instead of relying on the icon's bottom margin to push the title away from it, I'm now using flex with the column direction where the icon and the title are spaced apart.

This should easily allow for two-line-long database names.

![image](https://github.com/user-attachments/assets/50a25ed7-1ac9-4709-acb9-b376b91a5629)